### DISCUSSION TITLE
Master/Slave Reading

### DISCUSSION THREAD

**Discussion by sleepymaxx** [nathanmarlor/foxess_modbus#452] [Ideas]
Are you Guys (besides the option to add the second inverter als a new device as mentioned here:) 
https://github.com/nathanmarlor/foxess_modbus/discussions/406
planning to add the readings from a as slave connected inverter?
Explanation: i have a insallation of two invertes because of my greater overall pv output. The first is connected as a master the second is´t connected (over modbus) as the master is ruling all the readings and options from the slave...

**Comment by canton7** [comment]
Do you mean the CT2 current clamp? Yes, that's already exposed.

If not, how exactly is the slave connected to the master?

**Reply by sleepymaxx** [reply]
Master and slave are both each for itself connected normally (wired) to china.
Master is connected over modbus to the chint+connected over modbus to my home assistant (also i think its nececary to connect over the lan port in future because i will need the second mobus port for myPV integration) 
https://www.my-pv.com/en/products/ac-elwa-2/
Slave is (at the moment) not extra connected to my home assistant because all the readings and all the measurement also all the steerings ar coming over a direct can-bus (LAN-cable) between from the master: 
![pasted-from-clipboard](https://github.com/nathanmarlor/foxess_modbus/assets/103333137/9142a1b8-9628-48d3-88bf-ff1c3695ee95)

**Reply by FozzieUK** [reply]
When you are running as master/slave there is only one meter (on the master), the inverters resolve their loads via the canbus 'parallel' connection so they don't fight each as the load changes - they don't share any additional sensors other than what is essential to load balance.

If you want to see individual PV, Battery and Inverter voltage/current/power sensors the only way would be to connect the other inverter as a second device and monitor it in your Home Assistant.

**Comment by sleepymaxx** [comment]
Thank you... so ill stay only with a template sensor in HomeAssistant for my Energy Dashbord (only for grapics its enough...) teh readings and financial ouputs will be right because they´r coming from a different sensor.. so thanks anyway for your real great work... greetings from Germany... ;-)

**Comment by MartB** [comment]
Theoretically:

Wouldnt it be possible to hook up the slave as second device on the RS485 bus by chaining them together and providing another device id?

**Reply by iancg** [reply]
Yes - and I do this in practice too.

**Reply by canton7** [reply]
Yep, this is supported

**Reply by MartB** [reply]
Lovely, it works!
I just hooked this up and reconfigured everything.

Thanks for the confirmation :)

**Reply by KrisMH86** [reply]
I also have 2 inverters - do you just use an additional cable and extend the cable from pin 1 and 2 (I have 2 H3 inverters) from the master inverter to the slave?

Thank you for the help!

**Reply by williamjeccles** [reply]
I also have a USR modbus device per inverter with no issues.

**Reply by MartB** [reply]
@KrisMH86 Yeah just connect every "PIN 1" and "PIN 2" together after changing the modbus address for the slave.
